src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -226,8 +230,16 @@ public ObservableList<InternshipTask> getInternshipTaskList() {
         return internshipTasks.asUnmodifiableObservableList();
     }
 
+    /**
+     * Adds an internship task to the list of internship tasks, checking if the role exists first
+     * @param internshipTask
+     */
     public void addInternshipTask(InternshipTask internshipTask) {
-        internshipTasks.add(internshipTask);
+        if (!internshipRoles.contains(internshipTask.getInternshipRole())) {
+            logger.warning("Error adding internship task: internship role does not exist!");
```

Comment:
Do you want to do this check here, or in JsonAddressBook? Since all the error checking is there. This is really up to you. If dw to change can just merge it yourself. I will just approve this PR